erpnext/manufacturing/doctype/production_plan/production_plan.py

```diff
@@ -114,16 +123,26 @@ def get_so_items(self):
 		so_list = self.get_so_mr_list("sales_order", "sales_orders")
 
 		item_condition = ""
-		if self.item_code:
+		bom_item = "bom.item = so_item.item_code"
+		if self.item_code and frappe.db.exists('Item', self.item_code):
+			bom_item = self.get_bom_item()
```

Comment:
If none use `bom_item = "bom.item = so_item.item_code"`
```suggestion
			bom_item = self.get_bom_item() or bom_item
```